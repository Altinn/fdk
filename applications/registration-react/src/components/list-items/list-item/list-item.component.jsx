import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import localization from '../../../utils/localization';
import './list-item.scss';

export const ListItem = props => {
  const { title, status, path, statusNew } = props;

  if (!(title || status)) {
    return null;
  }

  const itemClass = cx('w-75', 'fdk-text-size-small', {
    'fdk-color2': !title
  });

  return (
    <div className="fdk-list-item d-flex">
      <Link className="w-100" to={path}>
        <div className="d-flex justify-content-between">
          <span className={itemClass}>
            {title || localization.listItems.missingTitle}
          </span>
          <div className="d-flex w-25 justify-content-between">
            {status === 'PUBLISH' && (
              <span>
                <i className="fa fa-circle fdk-color-cta mr-2 d-none d-sm-inline" />
                <span>{localization.listItems.statusPublished}</span>
              </span>
            )}
            {status === 'DRAFT' && (
              <span>
                <i className="fa fa-circle fdk-color3 mr-2 d-none d-sm-inline" />
                <span>{localization.listItems.statusDraft}</span>
              </span>
            )}
            {statusNew && (
              <span className="badge badge-pill badge-success">
                {localization.listItems.statusNew}
              </span>
            )}
          </div>
        </div>
      </Link>
    </div>
  );
};

ListItem.defaultProps = {
  title: null,
  status: null,
  path: null,
  statusNew: false
};

ListItem.propTypes = {
  title: PropTypes.string,
  status: PropTypes.string,
  path: PropTypes.string,
  statusNew: PropTypes.bool
};
