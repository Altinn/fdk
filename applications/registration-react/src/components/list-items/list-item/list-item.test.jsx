import React from 'react';
import { shallow } from 'enzyme';
import { ListItem } from './list-item.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    catalogId: '123'
  };
  wrapper = shallow(<ListItem {...defaultProps} />);
});

test('should render ListItem correctly with no props', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly "DRAFT" item', () => {
  const draftItem = {
    id: 1,
    title: {
      nb: 'Test item'
    },
    registrationStatus: 'DRAFT'
  };
  wrapper.setProps({
    item: draftItem
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly "PUBLISH" item', () => {
  const publishItem = {
    id: 1,
    title: {
      nb: 'Test item'
    },
    registrationStatus: 'PUBLISH'
  };
  wrapper.setProps({
    item: publishItem
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly when missing title', () => {
  const missingTitleItem = {
    id: 1,
    registrationStatus: 'DRAFT'
  };
  wrapper.setProps({
    item: missingTitleItem
  });
  expect(wrapper).toMatchSnapshot();
});
