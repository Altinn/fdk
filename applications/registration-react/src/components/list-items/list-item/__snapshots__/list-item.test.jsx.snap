// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem correctly "DRAFT" item 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-75 fdk-text-size-small"
      >
        Test item
      </span>
      <div
        className="d-flex w-25 justify-content-between"
      >
        <span>
          <i
            className="fa fa-circle fdk-color3 mr-2 d-none d-sm-inline"
          />
          <span>
            Utkast
          </span>
        </span>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly "PUBLISH" item 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-75 fdk-text-size-small"
      >
        Test item
      </span>
      <div
        className="d-flex w-25 justify-content-between"
      >
        <span>
          <i
            className="fa fa-circle fdk-color-cta mr-2 d-none d-sm-inline"
          />
          <span>
            Publisert
          </span>
        </span>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly when missing title 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-75 fdk-text-size-small fdk-color2"
      >
        Mangler tittel
      </span>
      <div
        className="d-flex w-25 justify-content-between"
      >
        <span>
          <i
            className="fa fa-circle fdk-color3 mr-2 d-none d-sm-inline"
          />
          <span>
            Utkast
          </span>
        </span>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly with no props 1`] = `""`;
